src/runtime/methodbinder.cs

```diff
@@ -475,26 +499,44 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw, MethodBase info, Meth
                     // XXX maybe better to do this before all the other rigmarole.
                     if (co == null)
                     {
+                        Exceptions.SetError(Exceptions.TypeError, "Invoked a non-static method with an invalid instance");
                         return null;
                     }
                     target = co.inst;
                 }
 
                 return new Binding(mi, target, margs, outs);
             }
-            // We weren't able to find a matching method but at least one
-            // is a generic method and info is null. That happens when a generic
-            // method was not called using the [] syntax. Let's introspect the
-            // type of the arguments and use it to construct the correct method.
-            if (isGeneric && info == null && methodinfo != null)
+            else if (isGeneric && info == null && methodinfo != null)
             {
+                // We weren't able to find a matching method but at least one
+                // is a generic method and info is null. That happens when a generic
+                // method was not called using the [] syntax. Let's introspect the
+                // type of the arguments and use it to construct the correct method.
                 Type[] types = Runtime.PythonArgsToTypeArray(args, true);
                 MethodInfo mi = MatchParameters(methodinfo, types);
-                return Bind(inst, args, kw, mi, null);
+                if (mi != null)
+                {
+                    return Bind(inst, args, kw, mi, null);
+                }
+            }
+            if (mismatchedMethods.Count == 1)
+            {
+                mismatchedMethods[0].Exception.Restore();
+            }
+            else if (mismatchedMethods.Count > 1)
+            {
+                var aggregateException = GetAggregateException(mismatchedMethods);
+                Exceptions.SetError(aggregateException);
             }
             return null;
         }
 
+        static AggregateException GetAggregateException(IEnumerable<MismatchedMethod> mismatchedMethods)
+        {
+            return new AggregateException(mismatchedMethods.Select(m => new ArgumentException($"{m.Exception.Message} in method {m.Method}")));
```

Comment:
Unfortunately, this causes test_method_parameters_change to fail (see the build results)

Reply:
I think you should disable the test for now. I dug around a bit, and it seems like domain reload would fail when a .NET object pointed to by Python has in turn references to Python objects (e.g. has either `PythonException` which causes `SerializationException` like in this case or `PyObject`, which would cause a refcount disbalance).

@BadSingleton and/or @benoithudson can later fix this issue and reenable the test case.